service/controller/resource/tccpn/template/template_main_auto_scaling_group.go

```diff
@@ -50,7 +50,7 @@ const TemplateMainAutoScalingGroup = `
 
         # Minimum amount of nodes that must always be running during a rolling
         # update.
-        MinInstancesInService: 1
+        MinInstancesInService: 0
```

Comment:
`MinInstancesInService` must be less than `Min`. Not sure how this didn't come up yet. This is the only solution I could think of, but I would like someone else to confirm that it makes sense. I don't know much about the master upgrade process.

Reply:
@calvix what are your thoughts on this? That may be fine with single master setups but in case a multi master setup uses this and 2 masters go down at the same time the etcd cluster is done.

Reply:
Small correction to my previous comment: the error message is `MinInstancesInService must be less than the autoscaling group's MaxSize` so we can either set `MinInstancesInService` to `MinSize - 1` (0 for single master, 2 for multi master) or increase `MaxSize` by 1 (2 for single master, 4 for multi master). I think the former makes sense--there has to be downtime while the single master is switched over so there's no reason to spin up a second master concurrently IMO.

Reply:
Our idea so far was to have 3 ASGs with 1 master each. There was a whole lot of discussion around this with Vaclav and Timo some time ago.

Reply:
Oh okay. I'm not sure about the implications in that case. Should we try to get it right now or go with this and fix HA later?

Reply:
I cannot decide this on my own now. We need to talk about this with Vaclav. Are you good to go by reverting the change or what is the exact problem you are facing?

Reply:
Yes, this is a good catch. I fucked up it when migrated to aws launch template from launch configuration